### PR TITLE
feat(forge): print compile time

### LINKED ISSUE #1125: Show compilation time
### Component

Forge

### Describe the feature you would like

Print time spent compiling, similar to test block summaries:

```
$ forge build
[⠒] Compiling...
[⠃] Compiling 35 files with 0.6.12
Compiler run successful. Finished in 782.40ms
```

### Additional context

We can add duration to [`ProjectCompileOutput`](https://github.com/gakonst/ethers-rs/blob/919a570237b0d92cfa34737d8803514caaa62612/ethers-solc/src/compile/output.rs#L17) and [`OutputDiagnostics`](https://github.com/gakonst/ethers-rs/blob/919a570237b0d92cfa34737d8803514caaa62612/ethers-solc/src/compile/output.rs#L356) in `ethers-rs`. Should we simply add this to the `ethers-rs` output message?

[TG discussion](https://t.me/foundry_rs/15349)

### PR DESCRIPTION
**Pull request by ecmendenhall**
## Motivation

Resolves #1125 

Print time spent compiling the project.

t-rec_3

## Solution

Print duration passed to `SpinnerReporter`. Depends on [gakonst/ethers-rs/#1098](https://github.com/gakonst/ethers-rs/pull/1098). 

This does look a little odd: when there is a lot of compiler output to display, the spinner will keep spinning for a few seconds, even though compilation is finished. (See the GIF above).

### CODE REVIEW

File: cli/src/term.rs [hunk 1]
```diff
@@ -228,8 +229,15 @@ impl Reporter for SpinnerReporter {
...
-    fn on_solc_success(&self, _solc: &Solc, _version: &Version, output: &CompilerOutput) {
+    fn on_solc_success(
+        &self,
+        _solc: &Solc,
+        _version: &Version,
+        output: &CompilerOutput,
+        duration: &Duration,
+    ) {
         self.solc_io_report.log_compiler_output(output);
+        self.send_msg(format!("Finished in {:.2?}", duration));
```
Comment (mattsse) [inline cli/src/term.rs:240]: Should we make it clearer that this is just the time `solc` took to compile, like "Solc finished in", perhaps?

**Review by mattsse — COMMENTED** [review]
great!
needs ethers bump.

could you try running in release mode (--release) to see how much time the spinner actually spents displaying "Finished".

**Comment by ecmendenhall** [comment]
> could you try running in release mode (--release) to see how much time the spinner actually spents displaying "Finished".

Ah, much better with the `--release` flag:

t-rec_6

Can you walk me through what needs to happen to bump the ethers version? Is this a breaking change to the reporter API?

**Review by gakonst — APPROVED** [review]
ACK, you need to "cargo update -p ethers" and remove the patch, otherwise good to go

**Pull request merged**